src/element/types.ts

```diff
@@ -13,10 +13,17 @@ type _ExcalidrawElementBase = Readonly<{
   width: number;
   height: number;
   seed: number;
+}>;
+
+export type Versioned<TElement extends _ExcalidrawElementBase> = TElement & {
   version: number;
   versionNonce: number;
   isDeleted: boolean;
-}>;
+};
+
+export type NonDeleted<TElement extends _ExcalidrawElementBase> = TElement & {
```

Comment:
I'd extend `ExcalidrawElement` here instead, for better readability in the codebase.

Reply:
`ExtantExcalidrawElement` as a name perhaps?

Reply:
No, I simply meant `<TElement extends ExcalidrawElement>` instead of `extends _ExcalidrawElementBase`, because the latter type is low-level and unnecessary to use here (and when it's read outside the module e.g. via intellisense, it will not make much sense).